driver/kubedriver.go

```diff
@@ -281,20 +281,32 @@ func (driver *KubeDriver) formResultSecret(r *base.ExperimentResult) (*corev1.Se
 // createExperimentResultSecret creates the experiment result secret
 func (driver *KubeDriver) createExperimentResultSecret(r *base.ExperimentResult) error {
 	if sec, err := driver.formResultSecret(r); err == nil {
-		secretsClient := driver.Clientset.CoreV1().Secrets(driver.Namespace())
-		_, err1 := secretsClient.Create(context.Background(), sec, metav1.CreateOptions{})
-		// TODO: Evaluate if result secret creation requires retries.
-		// Probably not. A get call precedes creation,
-		// and is retried if there are permission issues.
+		err1 := retry.OnError(
+			wait.Backoff{
+				Steps:    int(secretTimeout / retryInterval),
+				Cap:      secretTimeout,
+				Duration: retryInterval,
+				Factor:   1.0,
+				Jitter:   0.1,
+			},
+			func(err2 error) bool { // retry on specific failures
+				return kerrors.ReasonForError(err2) == metav1.StatusReasonForbidden
+			},
+			func() error {
+				secretsClient := driver.Clientset.CoreV1().Secrets(driver.Namespace())
+				_, err3 := secretsClient.Create(context.Background(), sec, metav1.CreateOptions{})
+				return err3
+			},
+		)
 		if err1 != nil {
-			err2 := fmt.Errorf("unable to create secret %v", sec.Name)
-			log.Logger.WithStackTrace(err1.Error()).Error(err2)
-			return err2
+			err4 := fmt.Errorf("unable to create secret %v", sec.Name)
```

Comment:
Is it good practice to enumerate error variables like this? Seeing it go to 4 makes me more and more concerned. Other than that, everything else looks good.

Reply:
I started by reusing error variable names ... until it started leading to subtle bugs. Given that readability is not impacted in anyway, I feel this is alright.